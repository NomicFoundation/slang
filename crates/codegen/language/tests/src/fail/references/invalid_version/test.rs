#![allow(unused_crate_dependencies)]

codegen_language_macros::compile!(Language(
    name = Foo,
    root_item = Bar,
    leading_trivia = Sequence([]),
    trailing_trivia = Sequence([]),
    versions = ["1.0.0", "2.0.0", "3.0.0"],
    sections = [Section(
        title = "Section One",
        topics = [Topic(
            title = "Topic One",
            items = [
                Struct(
                    name = Bar,
                    fields = (
                        field_1 = Optional(
                            // should have been disabled in "3.0.0"
                            kind = Terminal([Baz]),
                            enabled_in = "2.0.0"
                        ),
                        field_2 = Optional(
                            // should have been enabled in "2.0.0"
                            kind = Terminal([Baz]),
                            disabled_in = "3.0.0"
                        ),
                        field_3 = Optional(
                            // should have been enabled in "2.0.0" and disabled in "3.0.0"
                            kind = Terminal([Baz])
                        ),
                        field_4 = Optional(
                            // correct
                            kind = Terminal([Baz]),
                            enabled_in = "2.0.0",
                            disabled_in = "3.0.0"
                        )
                    )
                ),
                Token(
                    name = Baz,
                    definitions = [TokenDefinition(
                        enabled_in = "2.0.0",
                        disabled_in = "3.0.0",
                        scanner = Atom("baz")
                    )]
                )
            ]
        )]
    )]
));

fn main() {}
