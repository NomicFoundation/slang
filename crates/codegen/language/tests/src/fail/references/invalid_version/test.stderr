error: Reference 'Baz' is not defined in versions '3.0.0..MAX'.
  --> src/fail/references/invalid_version/test.rs:19:46
   |
19 | ...                   kind = Terminal([Baz]),
   |                                        ^^^

error: Reference 'Baz' is not defined in versions '1.0.0..2.0.0'.
  --> src/fail/references/invalid_version/test.rs:24:46
   |
24 | ...                   kind = Terminal([Baz]),
   |                                        ^^^

error: Reference 'Baz' is not defined in versions '1.0.0..2.0.0 | 3.0.0..MAX'.
  --> src/fail/references/invalid_version/test.rs:29:46
   |
29 | ...                   kind = Terminal([Baz])
   |                                        ^^^
